hail/src/main/scala/is/hail/asm4s/Code.scala

```diff
@@ -170,6 +170,22 @@ object Code {
     }
   }
 
+  def switch(target: Code[Int], dflt: Code[Unit], cases: Seq[(Int, Code[Unit])]): Code[Unit] = {
```

Comment:
Any reason not to make this
```scala
def switch[T](target: Code[Int], dflt: Code[T], cases: Seq[(Int, Code[T])]): Code[T]
```
?

Reply:
i think you run into some `TypeInfo` problems ... I could try to make it work

Reply:
to be clear i don't actually use `Code.switch` anywhere so i'm not very keen on making this any more general

Reply:
I think it's okay to leave it as is, I was just curious.

Reply:
actually it seems to work fine.

Reply:
Great!